### PR TITLE
Execute manip plan

### PR DESCRIPTION
**Pull request by FabianEP11**
- Added a server to get the namedTargetPoses of the planning groups. It helps users to command the robot to predefined poses using the robot manager interface.
- Added 3 attempts to execute last plan, and throw an error in case it fails.

### CODE REVIEW

File: include/temoto_robot_manager/robot.h [hunk 1]
```diff
@@ -57,6 +57,9 @@ class Robot : public temoto_core::BaseSubsystem
   void executeManipulationPath();
   
   geometry_msgs::Pose getManipulationTarget();
+
+  std::vector<std::string> getNamedTargetPoses(std::string& planning_group_name);
```
Comment (RValner) [inline include/temoto_robot_manager/robot.h]: shouldn't it be const ref?

File: src/robot.cpp [hunk 2]
```diff
@@ -608,15 +608,29 @@ void Robot::executeManipulationPath()
       planning_groups_.find(planning_group_name);  ///< Will throw if group does not exist
   if (group_it != planning_groups_.end())
   {
-    bool success;
+    bool success = false;
     group_it->second->setStartStateToCurrentState();
     group_it->second->setRandomTarget();
-    success = static_cast<bool>(group_it->second->execute(last_plan));
+    //success = static_cast<bool>(group_it->second->execute(last_plan));
```
Comment (RValner) [inline src/robot.cpp]: I guess you should remove it completely instead of leaving it as a comment

File: src/robot.cpp [hunk 3]
```diff
@@ -608,15 +608,29 @@ void Robot::executeManipulationPath()
...
+    size_t i=0;
+    while (!success && i<3)
```
Comment (RValner) [inline src/robot.cpp]: why not  `for (size_t i=0; !success || i<3; i++) {  ...  }`? just a bit more compact

File: src/robot.cpp [hunk 4]
```diff
@@ -641,6 +655,28 @@ geometry_msgs::Pose Robot::getManipulationTarget()
...
+std::vector<std::string> Robot::getNamedTargetPoses(std::string& planning_group_name)
+{
+  if (!planning_groups_.size())
+  {
+    throw CREATE_ERROR(temoto_core::error::Code::ROBOT_PLAN_FAIL,"Robot has no planning groups.");
+  }
+
+  FeatureManipulation& ftr = config_->getFeatureManipulation();
+
+  planning_group_name = (planning_group_name == "") ? ftr.getActivePlanningGroup() : planning_group_name;
```
Comment (RValner) [inline src/robot.cpp]: use `planning_group_name.empty()` instead of `planning_group_name == ""`

File: src/robot.cpp [hunk 5]
```diff
@@ -641,6 +655,28 @@ geometry_msgs::Pose Robot::getManipulationTarget()
   return current_pose;  
 }
 
+std::vector<std::string> Robot::getNamedTargetPoses(std::string& planning_group_name)
```
Comment (RValner) [inline src/robot.cpp]: like mentioned above, use const ref. i do see that you are doing some planing group name retrieval logic, but in that case just create a local variable for it. we don't want to actually return anything via the `planning_group_name` argument right?

**Review by RValner — CHANGES_REQUESTED** [review]
Minor requests but still have a look.

File: src/robot.cpp (hunk 5 above)
Comment (FabianEP11) [reply src/robot.cpp]: Ok, 
Added const to the planning_group_name argument

**Review by RValner — APPROVED** [review]
Good to go.

**Pull request merged**